Vtiger/Model/Validator/Constraints/DateValidator.php

```diff
@@ -21,31 +21,28 @@ class DateValidator extends ConstraintValidator
     public function validate($value, Constraint $constraint)
     {
         if (!$constraint instanceof Date) {
-            throw new UnexpectedTypeException($constraint, __NAMESPACE__ . '\Date');
+            throw new UnexpectedTypeException($constraint, Date::class);
         }
 
+        // custom constraints should ignore null and empty values to allow
+        // other constraints (NotBlank, NotNull, etc.) take care of that
         if (null === $value || '' === $value) {
             return;
         }
 
+        // check if item can be converted to string
         if (!is_scalar($value) && !(is_object($value) && method_exists($value, '__toString'))) {
             throw new UnexpectedTypeException($value, 'string');
         }
 
-        $date         = \DateTime::createFromFormat('Y-m-d', $value);
+        $date = DateTime::createFromFormat('Y-m-d', $value);
```

Comment:
Please see comment above.
```suggestion
        $date = \DateTime::createFromFormat('Y-m-d', $value);
```

Reply:
I am worried there are no testing steps. It is impossible to reproduce the bug because the bug is always in ''else' part which never happens.

Reply:
Even though we cannot replicate it we can still test that the sync works :)

Reply:
If you like to test the Vtiger CRM plugin, please use the credentials from secure.mautic.com.